File: Python3/프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    idx = 0
    mode = 0
    for i in code:
        if mode == 0:
            if i != str(1):
                if idx % 2 == 0:
                    answer += i
                    idx += 1
                else:
                    idx += 1
            elif i == str(1):
                mode = 1
                idx += 1
                    
        elif mode == 1:
            if i != str(1):
                if idx % 2 > 0:
                    answer += i
                    idx += 1
                else:
                    idx += 1
            elif i == str(1):
                mode = 0
                idx += 1
                
    if answer == '':
        answer = "EMPTY"
            
            
    return answer

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String answer = "";
        int mode = 0;
        char ch = ' ';
        
        for (int i = 0; i < code.length(); i++) {
            ch = code.charAt(i);
            if (mode == 0) {
                if (ch == '1') {
                    mode = 1;
                }
                else {
                    if (i % 2 == 0) {
                        answer += ch;
                    }
                }
            }
            else if (mode == 1) {
                if (ch == '1') {
                    mode = 0;
                }                
                else {
                    if (i % 2 > 0) {
                        answer += ch;
                    }
                }
            }
        }
        if (answer == "") {
            return "EMPTY";
        }
        else return answer;
    }
}